eng/testing/tests.wasm.targets

```diff
@@ -30,8 +30,10 @@
 
     <_AppArgs Condition="'$(WasmTestAppArgs)' != ''">$(_AppArgs) $(WasmTestAppArgs)</_AppArgs>
 
-    <RunScriptCommand Condition="'$(OS)' != 'Windows_NT'">$HARNESS_RUNNER $(_XHarnessArgs) %24WasmXHarnessArgs -- $(WasmXHarnessMonoArgs) %24WasmXHarnessMonoArgs $(_AppArgs) %24WasmTestAppArgs</RunScriptCommand>
-    <RunScriptCommand Condition="'$(OS)' == 'Windows_NT'">%HARNESS_RUNNER% $(_XHarnessArgs) %WasmXHarnessArgs%  -- $(WasmXHarnessMonoArgs) %WasmXHarnessMonoArgs% $(_AppArgs) %WasmTestAppArgs%</RunScriptCommand>
+    <_MonoArgs Condition="'$(RunAOTCompilation)' == 'true'">--setenv=CODE_GEN_MODE=aot</_MonoArgs>
```

Comment:
Instead of doing this here, it would make more sense to put https://github.com/dotnet/runtime/blob/01b7e73cd378145264a7cb7a09365b41ed42b240/src/mono/wasm/runtime/driver.c#L507

Using mono_eg_setenv() so that it is set in a way that doesn't depend on the runner.